Sources/Foundation/FileManager+Win32.swift

```diff
@@ -13,6 +13,96 @@
 import let WinSDK.INVALID_FILE_ATTRIBUTES
 import WinSDK
 
+extension URL {
+    fileprivate var NTPath: String {
+        "\\\\?\\\(CFURLCopyFileSystemPath(CFURLCopyAbsoluteURL(_cfObject), kCFURLWindowsPathStyle)!._swiftObject)"
+    }
+
+    fileprivate func withUnsafeNTPath<Result>(_ body: (UnsafePointer<WCHAR>) throws -> Result) rethrows -> Result {
+        return try NTPath.withCString(encodedAs: UTF16.self, body)
+    }
+}
+
+
+private func withNTPathRepresentation<Result>(of path: String, _ body: (UnsafePointer<WCHAR>) throws -> Result) throws -> Result {
+    func withNormalizedRepresentation<Result>(of path: String, _ body: (UnsafePointer<WCHAR>) throws -> Result) rethrows -> Result {
+        var path = path
+
+        // Strip the leading `/` on a RFC8089 path (`/[drive-letter]:/...` ).  A
+        // leading slash indicates a rooted path on the drive for teh current
+        // working directory.
+        if path.count > 3 {
+            let index0 = path.startIndex
+            let index1 = path.index(path.startIndex, offsetBy: 1)
+            let index2 = path.index(path.startIndex, offsetBy: 2)
+
+            if path[index0] == "/", path[index1].isLetter, path[index2] == ":" {
+                path.removeFirst()
+            }
+        }
+
+        // Win32 APIs can support `/` for the arc separator. However,
+        // symlinks created with `/` do not resolve properly, so normalize
+        // the path.
+        path = path.replacing("/", with: "\\")
+
+        // Droop trailing slashes unless it follows a drive specification.  The
+        // trailing arc separator after a drive specifier iindicates the root as
+        // opposed to a drive relative path.
+        while path.count > 1, path[path.index(before: path.endIndex)] == "\\",
```

Comment:
I think that I prefer the existing unnecessarily verbose spelling because it matches the following 2 cases and its easier to then realize what is going on.  I do agree with you though that `.last` is nicer.

Reply:
They're all style nits anyhow, it's always up to you 🙂